apollo-router/src/layers/async_checkpoint.rs

```diff
@@ -63,86 +61,11 @@ where
     fn layer(&self, service: S) -> Self::Service {
         AsyncCheckpointService {
             checkpoint_fn: Arc::clone(&self.checkpoint_fn),
-            inner: service,
+            service,
         }
     }
 }
 
-/// [`Service`] for OneShot (single use) Asynchronous Checkpoints. See [`ServiceBuilderExt::oneshot_checkpoint_async()`](crate::layers::ServiceBuilderExt::oneshot_checkpoint_async()).
-#[allow(clippy::type_complexity)]
-pub struct OneShotAsyncCheckpointService<S, Fut, Request>
```

Comment:
This has been removed because it doesn't work this service expects only to be called once. We are now reusing services.

It's pub so we need to add it to the migration guide.